src/resumable.ts

```diff
@@ -650,37 +743,46 @@ export class Resumable extends ResumableEventHandler {
    * Start or resume the upload of the provided files by initiating the upload of the first chunk
    */
   upload(): void {
+    Helpers.printDebugLow(this.debugVerbosityLevel, 'Starting Upload...');
     // Make sure we don't start too many uploads at once
-    if (this.isUploading) return;
+    if (this.isUploading) {
+      Helpers.printDebugLow(this.debugVerbosityLevel, 'Already uploading. Not starting again.');
+      return;
+    }
     // Kick off the queue
     this.fire('uploadStart');
     for (let num = 1; num <= this.simultaneousUploads; num++) {
       this.uploadNextChunk();
```

Comment:
To debug here, I'd appreciate a message indicating the current "worker number", e.g.

```suggestion
    for (let num = 1; num <= this.simultaneousUploads; num++) {
      Helpers.printDebugHigh(this.debugVerbosityLevel, 'Starting simultaneous upload ' + String(num) + ' / ' + String(this.simultaneousUploads) + ' ...');
      this.uploadNextChunk();
```

Reply:
Done. I removed the `String()` casts, because the numbers are just simple ints and we're still doing Javascript! 😝 
![image](https://github.com/pointcloudtechnology/resumable.js/assets/3686982/4b698d50-ec35-48aa-9c11-91887c05fc99)

I additionally added a corresponding `Started simultaneous upload` message after the call to `uploadNextChunk()`.